### PR TITLE
Fix data directory synchronization

### LINKED ISSUE #360: handle `task` workdir file deletions
Delete files from bucket when deleted on remote instance (so `terraform destroy` doesn't download them)

### LINKED ISSUE #376: `task` destroy is not always syncing all the data
I have identified this on Azure.
All the data seems not to be synced back always, however all the data seems to be within the task instance.

### Terraform

```tf
terraform {
    required_providers {
    iterative = {
        source = "iterative/iterative",
    }
    }
}

provider "iterative" {}

resource "iterative_task" "task" {
    cloud     = "az"
    machine   = "Standard_D2S_v3"
    region    = "us-west"
    spot      = 0

    name      = "david-mnist"
    workdir {
    input = "."
    }

    environment = {
    EPOCHS = 1
    }

    script = <<-END
    #!/bin/bash
    sudo apt update
    sudo apt-get install -y software-properties-common build-essential python3-pip 
    pip3 install -r requirements.txt
    python3 train.py

    ls output
    END
}
```

Fragment of TPI logs after calling `ls output`

```sh
Jan 26 13:37:21 tpi000001 tpi-task[8101]: confusion_matrix.png
Jan 26 13:37:21 tpi000001 tpi-task[8101]: metrics.json
Jan 26 13:37:21 tpi000001 tpi-task[8101]: seq.h5
Jan 26 13:37:21 tpi000001 tpi-task[8101]: seq.h5.epoch
Jan 26 13:37:21 tpi000001 tpi-task[8101]: tblogs
```

### destroy 

```sh
Plan: 0 to add, 0 to change, 1 to destroy.
iterative_task.task: Destroying... [id=tpi-david-mnist-59zgg25s-68txlw4t]
iterative_task.task: Still destroying... [id=tpi-david-mnist-59zgg25s-68txlw4t, 10s elapsed]
iterative_task.task: Still destroying... [id=tpi-david-mnist-59zgg25s-68txlw4t, 20s elapsed]
iterative_task.task: Still destroying... [id=tpi-david-mnist-59zgg25s-68txlw4t, 30s elapsed]
iterative_task.task: Still destroying... [id=tpi-david-mnist-59zgg25s-68txlw4t, 40s elapsed]
iterative_task.task: Still destroying... [id=tpi-david-mnist-59zgg25s-68txlw4t, 50s elapsed]
iterative_task.task: Still destroying... [id=tpi-david-mnist-59zgg25s-68txlw4t, 1m0s elapsed]
iterative_task.task: Still destroying... [id=tpi-david-mnist-59zgg25s-68txlw4t, 1m10s elapsed]
iterative_task.task: Still destroying... [id=tpi-david-mnist-59zgg25s-68txlw4t, 1m20s elapsed]
iterative_task.task: Destruction complete after 1m30s

Destroy complete! Resources: 1 destroyed.
```

### `ls output` locally
 
```
tblogs
``` 

Only folder tblogs has been synced back 😞

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
- Closes #360
 - Closes #376

### CODE REVIEW

**Comment by DavidGOrtega** [comment]
Unfortunately its not fixing #376

**Comment by 0x2b3bfa0** [comment]
Discovered that #322 was broken and some commits didn't get pushed. 🙈

**Comment by 0x2b3bfa0** [comment]
@DavidGOrtega, please try again with https://github.com/iterative/terraform-provider-iterative/pull/377/commits/04db00a00faf0cd11b3d8c1209b83524bde041e6; it should fix #376 now.

File: task/common/machine/script.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ sudo tee /usr/bin/tpi-task-shutdown << 'END'
 if ! test -z "$CI"; then
   cml rerun-workflow
 fi
+sleep 20; while pgrep rclone > /dev/null; do sleep 1; done
```
Comment (0x2b3bfa0) [inline task/common/machine/script.go:40]: Note: that `sleep 20` is not magical, just two times the `sleep 10` from [line 118](https://github.com/iterative/terraform-provider-iterative/pull/377/files#diff-fd1d8ac8ab64f155f5b5a078ae744a577707c4254732ae35597facadd0acdf86R118); ask Nyquist, Shannon and a few others why. 😄

**Review by DavidGOrtega — APPROVED** [review]

**Pull request merged**